cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -106,9 +97,9 @@
   private final DatasetFramework datasetFramework;
   private final SecureStore secureStore;
   private final TransactionSystemClient transactionSystemClient;
-  private final ConcurrentMap<ApplicationId, Injector> appInjectors;
   private final Path previewDataDir;
   private final PreviewRunnerModuleFactory previewRunnerModuleFactory;
+  private Injector previewInjector;
```

Comment:
Question - here if the preview service gets restarted, is there a way to restore the previous preview runs? Previously when using a map, we will scan the folder names to restore the previous runs, how is this part of logic get handled now?

Reply:
The restoration in the previous code was primarily to start preview application specific services required to serve the preview data for that particular application. In the new code since we have single table for all preview runs we don't need to startup preview application specific services. What is missing is ability to mark orphan previews to be KILLED as here https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewRunner.java#L332. Have files bug to tackle that - https://issues.cask.co/browse/CDAP-16997